litestar/openapi/plugins.py

```diff
@@ -301,6 +302,79 @@ def render(self, request: Request, openapi_schema: dict[str, Any]) -> bytes:
         )
 
 
+class ScalarRenderPlugin(OpenAPIRenderPlugin):
+    """Plugin to render an OpenAPI schema using Scalar.
+
+    .. versionadded:: 2.5.0
+    """
+
+    def __init__(
+        self,
+        *,
+        version: str = "1.8.0",
+        js_url: str | None = None,
+        css_url: str | None = None,
+        path: str | Sequence[str] = "/scalar",
+        **kwargs: Any,
+    ) -> None:
+        """Initialize the Scalar OpenAPI UI render plugin.
+
+        Args:
+            version: Scalar version to download from the CDN.
+                If js_url is provided, this is ignored.
+            js_url: Download url for the Scalar JS bundle.
+                If not provided, the version will be used to construct the url.
+            css_url: Download url for the Scalar CSS bundle.
+                If not provided, the Litestar-provided CSS will be used.
+            path: Path to serve the OpenAPI UI at.
+            **kwargs: Additional arguments to pass to the base class.
+        """
+        self.js_url = js_url or f"https://cdn.jsdelivr.net/npm/@scalar/api-reference@{version}"
+        self.css_url = css_url or "_styles/scalar.css"
```

Comment:
We can probably leverage the `OpenAPIRenderPlugin.receive_router()` callback to register a handler that serves our custom css if another URL hasn't already been registered.

Reply:
Done